### ISSUE TITLE
Unit Testing - Task Plugins

### ISSUE THREAD

**Issue by jtsextonMITRE** [usnistgov/dioptra#96]
This is the list of functions in the task-plugins folder that will need unit tests. Only exposed functions have been selected, except for the metrics which should all be tested..

**Artifacts:**
- [ ] [`download_all_artifacts_in_run` in dioptra_builtins/artifacts/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/mlflow.py)
- [ ] [`upload_data_frame_artifact` in dioptra_builtins/artifacts/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/mlflow.py)
- [ ] [`upload_directory_as_tarball_artifact` in dioptra_builtins/artifacts/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/mlflow.py)
- [ ] [`upload_file_as_artifact` in dioptra_builtins/artifacts/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/mlflow.py)
- [ ] [`extract_tarfile` in dioptra_builtins/artifacts/utils.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/utils.py)
- [ ] [`make_directories` in dioptra_builtins/artifacts/utils.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/artifacts/utils.py)

**Attacks:**
- [ ] [`create_adversarial_fgm_dataset` in dioptra_builtins/attacks/fgm.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/attacks/fgm.py)

**Data:**
- [ ] [`create_image_dataset` in dioptra_builtins/data/tensorflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/data/tensorflow.py)
- [ ] [`get_n_classes_from_directory_iterator` in dioptra_builtins/data/tensorflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/data/tensorflow.py)

**Estimators:**
- [ ] [`init_classifier` in dioptra_builtins/estimators/keras_classifiers.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/estimators/keras_classifiers.py)
- [ ] [`fit` in dioptra_builtins/estimators/methods.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/estimators/methods.py)
- [ ] [`predict` in dioptra_builtins/estimators/methods.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/estimators/methods.py)

**Metrics:**
- [ ] [`get_distance_metric_list` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`get_distance_metric` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`l_inf_norm` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`l_1_norm` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`l_2_norm` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`paired_cosine_similarities` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`paired_euclidean_distances` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`paired_manhattan_distances` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`paired_wasserstein_distances` in dioptra_builtins/metrics/distance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/distance.py)
- [ ] [`get_performance_metric_list` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`get_performance_metric` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`accuracy` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`roc_auc` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`categorical_accuracy` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`mcc` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`f1` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`precision` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)
- [ ] [`recall` in dioptra_builtins/metrics/performance.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/metrics/performance.py)

**Random:**
- [ ] [`init_rng` in dioptra_builtins/random/rng.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/random/rng.py)
- [ ] [`draw_random_integer` in dioptra_builtins/random/sample.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/random/sample.py)
- [ ] [`draw_random_integers` in dioptra_builtins/random/sample.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/random/sample.py)

**Registry:**
- [ ] [`load_wrapped_tensorflow_keras_classifier` in dioptra_builtins/registry/art.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/registry/art.py)
- [ ] [`add_model_to_registry` in dioptra_builtins/registry/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/registry/mlflow.py)
- [ ] [`get_experiment_name` in dioptra_builtins/registry/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/registry/mlflow.py)
- [ ] [`load_tensorflow_keras_classifier` in dioptra_builtins/registry/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/registry/mlflow.py)

**Tracking:**
- [ ] [`log_metrics` in dioptra_builtins/tracking/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/tracking/mlflow.py)
- [ ] [`log_parameters` in dioptra_builtins/tracking/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/tracking/mlflow.py)
- [ ] [`log_tensorflow_keras_estimator` in dioptra_builtins/tracking/mlflow.py](https://github.com/usnistgov/dioptra/blob/main/task-plugins/dioptra_builtins/tracking/mlflow.py)
